src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -272,20 +308,25 @@ internal virtual void Validate()
             // Validate that we have a client id
             if (string.IsNullOrWhiteSpace(Config.ClientId))
             {
-                throw new InvalidOperationException(MsalErrorMessage.NoClientIdWasSpecified);
+                throw new MsalClientException(MsalError.NoClientId, MsalErrorMessage.NoClientIdWasSpecified);
             }
 
             //Adfs does not require client id to be in the form of a Guid
             if (Config.AuthorityInfo?.AuthorityType != AuthorityType.Adfs && !Guid.TryParse(Config.ClientId, out Guid clientIdGuid))
             {
-                throw new InvalidOperationException(MsalErrorMessage.ClientIdMustBeAGuid);
+                throw new MsalClientException(MsalError.ClientIdMustBeAGuid, MsalErrorMessage.ClientIdMustBeAGuid);
```

Comment:
>MsalClientException [](start = 26, length = 19)

I suppose this is kind of a breaking change... though it's technically more correct. @JM, thoughts on exposed exception types?

Reply:
Theoretically yes, exceptions are part of the contract, but this isn't enforced like in Java, it's more on us to make a judgement. 

I don't expect developers to write exception handling code for things like using a badly form client ID. MsalClient exception will give a better experience.